File: demo/src/main/java/com/ramo/test/MainActivity.java
package com.ramo.test;

import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import android.os.Bundle;
import android.view.View;
import android.widget.TextView;
import android.widget.Toast;

import com.ramo.sweetrecycleradapter.ISetItemView;
import com.ramo.sweetrecycleradapter.OnRecyclerItemClickListener;
import com.ramo.sweetrecycleradapter.SweetRecyclerAdapter;

import java.util.ArrayList;
import java.util.List;

public class MainActivity extends AppCompatActivity {

    List<ExampleModel> fakeList;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        fakeData();


        RecyclerView recyclerView = findViewById(R.id.rw);
        recyclerView.setLayoutManager(new LinearLayoutManager(this));

        SweetRecyclerAdapter<ExampleModel> sweetRecyclerAdapter = new SweetRecyclerAdapter<>(this,fakeList);

        sweetRecyclerAdapter.addHolder(R.layout.item1, 0, new ISetItemView<ExampleModel>() {
            @Override
            public void setItemView(View v, ExampleModel item) {
                TextView tw1 = v.findViewById(R.id.tw1);
                tw1.setText(item.getText());
            }
        }, new OnRecyclerItemClickListener<ExampleModel>() {
            @Override
            public void onRecyclerItemListener(View v, ExampleModel item) {
                Toast.makeText(MainActivity.this, "Layout0 on Clicked", Toast.LENGTH_SHORT).show();
            }
        });

        sweetRecyclerAdapter.addHolder(R.layout.item2, 1, new ISetItemView<ExampleModel>() {
            @Override
            public void setItemView(View v, ExampleModel item) {
                TextView tw1 = v.findViewById(R.id.l2tw1);
                tw1.setText(item.getText());
            }
        }, new OnRecyclerItemClickListener<ExampleModel>() {
            @Override
            public void onRecyclerItemListener(View v, ExampleModel item) {
                Toast.makeText(MainActivity.this, "Layout1 on Clicked", Toast.LENGTH_SHORT).show();
            }
        });

        recyclerView.setAdapter(sweetRecyclerAdapter);

    }


    private void fakeData()
    {
        fakeList =  new ArrayList<>();
        fakeList.add(new ExampleModel(0, "RamoL0"));
        fakeList.add(new ExampleModel(1, "RamoL1"));
        fakeList.add(new ExampleModel(0, "RamoL0"));
        fakeList.add(new ExampleModel(0, "RamoL0"));
        fakeList.add(new ExampleModel(1, "RamoL1"));
    }
}

File: demo/src/main/java/com/ramo/test/MainActivity.kt
package com.ramo.test

import android.os.Bundle
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.ramo.sweetrecycleradapter.SweetRecyclerAdapter
import com.ramo.test.databinding.Item1Binding
import com.ramo.test.databinding.Item2Binding

class MainActivity : AppCompatActivity() {
    var fakeList = mutableListOf<ExampleModel>()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        fakeData()
        val recyclerView = findViewById<RecyclerView>(R.id.rw)
        recyclerView.layoutManager = LinearLayoutManager(this)


        val sweetRecyclerAdapter = SweetRecyclerAdapter<ExampleModel>()
        sweetRecyclerAdapter.addHolder(R.layout.item1) { view, item ->
            val vb = Item1Binding.bind(view)
            vb.tw1.text = item.text
        }
        sweetRecyclerAdapter.addHolder(R.layout.item2) { view, item ->
            val vb = Item2Binding.bind(view)
            vb.l2tw1.text = item.text
        }

        sweetRecyclerAdapter.setOnItemClickListener { v, item ->
            Toast.makeText(this, "denemee", Toast.LENGTH_SHORT).show()
        }

        recyclerView.adapter = sweetRecyclerAdapter

        sweetRecyclerAdapter.submitList(fakeList)
    }

    private fun fakeData() {
        fakeList.add(ExampleModel(0, "RamoL0"))
        fakeList.add(ExampleModel(1, "RamoL1"))
        fakeList.add(ExampleModel(0, "RamoL0"))
        fakeList.add(ExampleModel(0, "RamoL0"))
        fakeList.add(ExampleModel(1, "RamoL1"))
    }
}
